src/main/java/com/hivemq/edge/modules/api/adapters/ProtocolAdapterCapability.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.edge.modules.api.adapters;
+
+import com.google.common.base.Preconditions;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+
+/**
+ * @author Simon L Johnson
+ */
+public interface ProtocolAdapterCapability {
+
+    /**
+     * can the adapter-type read values from the external source and publish them into the system
+     **/
+    byte READ = 0b00000001;
```

Comment:
yeah, we have byte here because we use the bit-wise or operations on them to compile multiple together.

Reply:
Uff in the end we created a List based on the bytes. So this absolutely makes no sense. Instead of byte manipulation we should just return a Set over this enum anyway.